// jikants/dist/src/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const got_1 = __importDefault(require("got"));
const p_memoize_1 = __importDefault(require("p-memoize"));
const p_queue_1 = __importDefault(require("p-queue"));
const pino_1 = __importDefault(require("pino"));
const quick_lru_1 = __importDefault(require("quick-lru"));
const package_json_1 = __importDefault(require("../package.json"));
exports.baseUrl = "https://api.jikan.moe/v3";
exports.queue = new p_queue_1.default({ concurrency: 2 });
const http = got_1.default.extend({
    baseUrl: exports.baseUrl,
    headers: {
        "User-Agent": `${package_json_1.default.name} / ${package_json_1.default.version} (${package_json_1.default.repository.url})`
    },
    json: true
});
exports.api = p_memoize_1.default(http, { cache: new quick_lru_1.default({ maxSize: 1000 }) });
exports.Logger = pino_1.default({
    name: "jikants",
    prettyPrint: true
});
